programs/jito-protecc/src/lib.rs

```diff
@@ -15,23 +16,54 @@ pub mod jito_protecc {
         Ok(())
     }
 
-    pub fn pre_guard(ctx: Context<PreGuard>, bump: u8) -> Result<()> {
+    pub fn pre_guard(
```

Comment:
should we split out sol + non-sol guards? this won't work for SOL bc SOL isn't token account

Reply:
can add another set of ixs specifically for SOL and keep this as is. I like the current api b/c it allows to optionally check token and SOL balance

Reply:
wfm. do you see a case where guarding lamports here makes sense then? can't think of one and maybe saves a little tx space if we remove it

Reply:
Main reason is SOL balance rug protecc since clients might blind sign

Reply:
wouldn't it just be the rent in the account?

Reply:
are you conflating guarded state with guarded account? the former is the PDA the latter can be any account

Reply:
SOL balance rug protecc on a token account would be rent (like 0.002 SOL or whatever it is)